### ISSUE TITLE
On-board aligner from "Huang et al., Less Peaky and More Accurate CTC Forced Alignment by Label Priors"

### ISSUE THREAD

**Issue by dmitry-mli** [pytorch/audio#3826]
### 🚀 The feature

Consider on-boarding aligner from [Huang et al., Less Peaky and More Accurate CTC Forced Alignment by Label Priors](https://arxiv.org/abs/2406.02560) (@huangruizhe) to the existing set of aligners given it improves alignment accuracy compared to the existing Wav2Vec2 CTC aligner by up to 60% P50 on English.

### Motivation, pitch

Today, torch audio offers [Forced Alignment](https://pytorch.org/audio/stable/tutorials/forced_alignment_tutorial.html) through a simple extendable interface. The recently published aligner [Huang et al., Less Peaky and More Accurate CTC Forced Alignment by Label Priors](https://arxiv.org/abs/2406.02560) ([github](https://github.com/huangruizhe/audio/blob/aligner_label_priors/examples/asr/librispeech_alignment/eval.py)) drives the word boundary error (WBE) down (better) compared to Wav2Vec2. We explored the model and had WBE for our English samples decrease by up to 60% for P50, 45% for P70 and 15% for P95 compared to Wav2Vec2 CTC alignment.

<img width="576" alt="image" src="https://github.com/user-attachments/assets/cf70da5c-ec57-402c-a4e9-81753db30f6f">

### Alternatives

This request is related to a particular research.

### Additional context

Thanks for consideration. @huangruizhe @jamesr66a

**Comment by huangruizhe** [comment]
Thanks for your interests in our work and sharing the nice results! As we have been switching between projects, things have been greatly delayed. Regarding the plan, I will be more available in late September and October. I will work on it at that time!

**Comment by dmitry-mli** [comment]
Looking forward to it, thank you!

**Comment by christincha** [comment]
A catch up here, if there is any updated plan for incorporating the [Huang et al., Less Peaky and More Accurate CTC Forced Alignment by Label Prior](https://arxiv.org/abs/2406.02560) to current Pytorch audio aligner!

**Comment by huangruizhe** [comment]
Hi @christincha, I am still working on it. Before making it official, if you hope to do any experiments, maybe check this out: https://colab.research.google.com/drive/1xciHB1Twi7VFutACrv94-Ejff1-VrjzL?usp=sharing
There are different implementations of the proposed CTC loss, the training recipe, visualization tools as well as a pretrained model.
